lib/archethic/contracts/interpreter/library.ex

```diff
@@ -217,4 +195,34 @@ defmodule Archethic.Contracts.Interpreter.Library do
       bin
     end
   end
+
+  @doc """
+  Get the genesis address of the chain
+  """
+  @spec get_genesis_address(binary()) ::
+          binary()
+  def get_genesis_address(address) do
+    addr_bin = decode_binary(address)
+    nodes = Election.chain_storage_nodes(address, P2P.authorized_and_available_nodes())
+
+    case TransactionChain.fetch_genesis_address_remotely(addr_bin, nodes) do
+      {:error, _} -> address
```

Comment:
I have an open question, here if we get an error while fetching the data, we return the address. So the smart contract will continue with the same address that may not be accurate.
Maybe we should crash the execution of the contract if there is an error while fetching data.
I think we should have the same behavior between all fetching data, so this may be the subject of a new issue, what do you think ?

Reply:
can we return just "error"

Reply:
Returning \"error\" means that the developer should always have a if statement to handle \"error\" or the expected result

Reply:
its better than crashing??

Reply:
we will have match case